drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -168,68 +301,116 @@ local function initialize_switch(driver, device)
-    local id = 0
-    for _, ep in ipairs(device.endpoints) do
-      -- main_endpoint only supports server cluster by definition of get_endpoints()
-      if main_endpoint == ep.endpoint_id then
-        for _, dt in ipairs(ep.device_types) do
-          -- no device type that is not in the switch subset should be considered.
-          if (ON_OFF_SWITCH_ID <= dt.device_type_id and dt.device_type_id <= ON_OFF_COLOR_DIMMER_SWITCH_ID) then
-            id = math.max(id, dt.device_type_id)
+
+  if device:get_field(IS_PARENT_CHILD_DEVICE) then
+    device:set_find_child(find_child)
+  end
+
+  if component_map_used then
+    device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
+  end
+
+  if #switch_eps > 0 then
+    -- The case where num_switch_server_eps > 0 is a workaround for devices that have a
+    -- Light Switch device type but implement the On Off cluster as server (which is against the spec
+    -- for this device type). By default, we do not support Light Switch device types because by spec these
+    -- devices need bindings to work correctly (On/Off cluster is client in this case), so these device types
+    -- do not have a generic fingerprint and will join as a matter-thing. However, we have seen some devices
+    -- claim to be Light Switch device types and still implement their clusters as server, so this is a
+    -- workaround for those devices.
+    if num_switch_server_eps > 0 and detect_matter_thing(device) == true then
+      local id = 0
+      for _, ep in ipairs(device.endpoints) do
+        -- main_endpoint only supports server cluster by definition of get_endpoints()
+        if main_endpoint == ep.endpoint_id then
+          for _, dt in ipairs(ep.device_types) do
+            -- no device type that is not in the switch subset should be considered.
+            if (ON_OFF_SWITCH_ID <= dt.device_type_id and dt.device_type_id <= ON_OFF_COLOR_DIMMER_SWITCH_ID) then
+              id = math.max(id, dt.device_type_id)
+            end
           end
+          break
         end
-        break
       end
+
+      if device_type_profile_map[id] ~= nil then
+        device:try_update_metadata({profile = device_type_profile_map[id]})
+      end
+    end
+  elseif #button_eps > 0 then
+    local battery_support = false
+    if device.manufacturer_info.vendor_id ~= HUE_MANUFACTURER_ID and
+      #device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.Feature.BATTERY}) > 0 then
+      battery_support = true
+    end
+    if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
+      if battery_support then
+        profile_name = string.format("%d-button-battery", #button_eps)
+      else
+        profile_name = string.format("%d-button", #button_eps)
+      end
+    elseif not battery_support then
+      -- a battery-less button/remote (either single or will use parent/child)
+      profile_name = "button"
+    end
+
+    if profile_name then
```

Comment:
Coudl combine this conditional with the one below

Reply:
Updated in 340df0f5c6c6d95d4d9b7c19e2b70b4b2fb646e2